.github/workflows/cypress.yml

```diff
@@ -60,6 +62,8 @@ jobs:
       - name: Cypress run
         uses: cypress-io/github-action@v2
         with:
+          parallel: ${{ env.CYPRESS_RECORD_KEY != 0 }}
+          group: ''
```

Comment:
I'm not 100% sure if `group` needs to be conditional or not, but I figured we can merge it into a test branch `cypress-parallel-test` and see.

@raisedadead any idea if you can do something like `${{ env.CYPRESS_RECORD_KEY != 0 ? '' : 'Group name' }}` in an expression?  I couldn't see examples of it, so I assume not.

Reply:
Here is what you have at your disposal: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions, however I can give a concrete example if you explain the context a little bit more. What is it we are trying to do here?

Reply:
So, https://github.com/cypress-io/github-action#parallel has a `group` parameter.  However, we don't want to have that set on PRs because it will try (and fail) to record.  We saw this here: https://github.com/Sembauke/freeCodeCamp/commit/93c252e78b3bddab607e860016bcd1c13006a7e9

It might be necessary to have that set conditionally to `''` on PRs and, say, `Parallel run` (something non-empty) when it's triggered by a push to main.

Reply:
In that case you should setup a workflow with all the events that you would want to trigger the workflow for at the top level. Then on you can create multiple jobs and use the [`github.context`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) to create conditionals within the jobs config to run a config. 

It's not DRY code per se, but will keep the configs together and readable.

Here is some sudo code as an example

```txt
name: Workflow Name
on: [push, pull_request]

jobs:
  if: ${{ github.event_name == 'push' }}
  cypress-job-name-on-push:
    name: cypress-job-name-on-push
    runs-on: ubuntu-18.04
    strategy:
      ...
      matrix:
        ...
    ...
    steps:
	  ...
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          parallel: true
          group: 'Running on push'
          ...
  if: ${{ github.event_name == 'pull_request' }}
  cypress-job-name-on-pull-request:
    name: cypress-job-name-on-pull-request
    runs-on: ubuntu-18.04
    strategy:
      ...
      matrix:
        ...
    ...
    steps:
	  ...
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          parallel: true
          group: ''
          ...

```

Reply:
That makes a lot of sense. How about we merge this code into the test branch as is (@raisedadead if you're okay with that).

Chances are we'll need to split it into PR and push, but we can do that after the test.